api/swagger.yml

```diff
@@ -2523,7 +2554,98 @@ paths:
           description: too many requests
         default:
           $ref: "#/components/responses/ServerError"
-
+  /auth/external/principals:
+    get: 
+      tags: 
+        - auth
+        - external
+      operationId: listExternalPrincipals
+      summary: list external principals
+      parameters: 
+        - $ref: "#/components/parameters/PaginationPrefix"
+        - $ref: "#/components/parameters/PaginationAfter"
+        - $ref: "#/components/parameters/PaginationAmount"
+      responses: 
+        200: 
+          description: external principal list
+          content: 
+            application/json: 
+              schema: 
+                $ref: "#/components/schemas/ExternalPrincipalList"
+        401: 
+          $ref: "#/components/responses/Unauthorized"
+        420: 
+          description: too many requests
+        default: 
+          $ref: "#/components/responses/ServerError"
+    post: 
+      tags: 
+        - auth
+        - external
+      operationId: createExternalPrincipal
+      summary: Create principal as external identity connected to lakeFS user
```

Comment:
I thought that external auth are configurable, not something the user need to make an API call to add

Reply:
I'm not sure what you mean, perhaps a miss understanding? 
As the summary suggests (maybe not that well?) it will create a specific principal. 
i.e `arn:123:role:abc` -> `user_a`.